plugins/modules/cml_link_node.py

```diff
@@ -0,0 +1,219 @@
+        lab: "{{ cml_lab }}"
+        source_node: "{{ source_node }}"
+        destination_node: "{{ destination_node }}"
+        action: create
+
+- name: Update a link between two CML nodes
+  hosts: cml_hosts
+  connection: local
+  gather_facts: no
+  tasks:
+    - name: Link nodes
+      cisco.cml.cml_link_node:
+        host: "{{ cml_host }}"
+        user: "{{ cml_username }}"
+        password: "{{ cml_password }}"
+        lab: "{{ cml_lab }}"
+        source_node: "{{ source_node }}"
+        destination_node: "{{ destination_node }}"
+        update_node: "{{ update_node }}"
+        action: update
+
+- name: Delete a link between two CML nodes
+  hosts: cml_hosts
+  connection: local
+  gather_facts: no
+  tasks:
+    - name: Link nodes
+      cisco.cml.cml_link_node:
+        host: "{{ cml_host }}"
+        user: "{{ cml_username }}"
+        password: "{{ cml_password }}"
+        lab: "{{ cml_lab }}"
+        source_node: "{{ source_node }}"
+        destination_node: "{{ destination_node }}"
+        action: delete
+"""
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible_collections.cisco.cml.plugins.module_utils.cml_utils import cmlModule, cml_argument_spec
+
+def run_module():
+    # define available arguments/parameters a user can pass to the module
+    argument_spec = cml_argument_spec()
+    argument_spec.update(
+        lab=dict(type='str', required=True, fallback=(env_fallback, ['CML_LAB'])),
+        action=dict(type='str', required=True),
+        source_node=dict(type='str', required=True),
+        destination_node=dict(type='str', required=True),
+        update_node=dict(type='str', required=False),
+        tags=dict(type='list', elements='str'),
+        x=dict(type='int'),
+        y=dict(type='int'),
+        wait=dict(type='bool', default=False),
+    )
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+    cml = cmlModule(module)
+
+    labs = cml.client.find_labs_by_title(cml.params['lab'])
+    if len(labs) > 0:
+        lab = labs[0]
+    else:
+        cml.fail_json("Cannot find lab {0}".format(cml.params['lab']))
+
+    # get both nodes by name
+    source_node = cml.get_node_by_name(lab, cml.params['source_node'])
+    destination_node = cml.get_node_by_name(lab, cml.params['destination_node'])
+    update_node = cml.get_node_by_name(lab, cml.params['update_node'])
+
+    if source_node == None or destination_node == None:
+        cml.fail_json("One or more nodes cannot be found. Nodes need to be created before a link can be established")
+        cml.exit_json(**cml.result)
+        return
+    
+    link = cml.get_link_by_nodes(lab, source_node, destination_node)
```

Comment:
virl2-client node supports this natively:
```python
link = source_node.get_link_to(destination_node)
```

as per:
https://github.com/CiscoDevNet/virl2-client/blob/22693cd1be5f321f3848fe591436524abdb804fe/virl2_client/models/node.py#L635
```python
        def get_link_to(self, other_node: Node) -> Link | None:
        """
        Return one link between this node and another.

        :param other_node: The other node.
        :returns: A link between this node and the other node, if one exists.
        """
        for link in self.links():
            if other_node in link.nodes:
                return link
        return None
```